lib/IR/LegacyPassManager.cpp

```diff
@@ -679,12 +679,37 @@ void PMTopLevelManager::schedulePass(Pass *P) {
     return;
   }
 
+  // HLSL Change - begin
+  class direct_stderr_stream : public raw_ostream {
+    uint64_t current_pos() const override { return 0; }
+    /// See raw_ostream::write_impl.
+    void write_impl(const char *Ptr, size_t Size) override {
+      fwrite(Ptr, Size, 1, stderr);
+    }
+  };
+
+  static direct_stderr_stream stderr_stream;
+  // HLSL Change - end
+
   if (PI && !PI->isAnalysis() && ShouldPrintBeforePass(PI)) {
     Pass *PP = P->createPrinterPass(dbgs(), std::string("*** IR Dump Before ") +
                                          P->getPassName().str() + " ***");
     PP->assignPassManager(activeStack, getTopLevelPassManagerType());
   }
 
+  // HLSL Change - begin
+  if (PI && !PI->isAnalysis() &&
+      (this->HLSLPrintBeforeAll ||
+       (this->HLSLPrintBefore.size() &&
+        this->HLSLPrintBefore.count(PI->getPassArgument())))) {
```

Comment:
It's a nit, but this conditional and the one below that it was modeled after are getting pretty hairy. The previous conditional also checks for `PI && !PI->isAnalysis()`. If we could couch both of these within two corresponding `if(PI && !PI->isAnalysis()` blocks, there would be that much more joy in my life 😄

Reply:
Given that the other conditional is really only there as a way to avoid conflicts with upstream (ShouldPrintXPass is stubbed out elsewhere) I'm not sure that it makes sense to keep it if we're going to modify it. That said, it's probably not really doing it's job so maybe it's best to just delete it either way.

Reply:
Refactored a bit and used a helper function to simplify this